src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,134 +2,162 @@ import log from 'electron-log';
 import React, { FC, useEffect, useRef } from 'react';
 import Col from 'react-bootstrap/Col';
 import Row from 'react-bootstrap/Row';
-import useInterval from 'react-use/lib/useInterval';
 import WaveSurfer from 'wavesurfer.js';
+import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone';
 import SpectrogramPlugin from 'wavesurfer.js/dist/plugin/wavesurfer.spectrogram';
 
 import { noOp } from '../../../common/env-and-consts';
-import { checkFileExistence, join, readFile } from '../../utils';
+import { acquireAudioInputStream, checkFileExistence, join, readFile } from '../../utils';
+import { FileMonitor } from '../../utils/file-monitor';
+import { useAudioInputOutputDevices } from '../settings-page/hooks';
 
 import { State } from './types';
 
 interface RecordingVisualizationProps {
   prevState?: State;
   state: State;
   filePath?: string;
+  basePath: string;
+  recItemName: string;
 }
 
 const GRAPH_WIDTH = '60%';
-const GRAPH_HEIGHT = 80;
-const EMPTY_IMAGE_URL = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
 
-export const RecordingVisualization: FC<RecordingVisualizationProps> = ({ prevState, state, filePath }) => {
+export const RecordingVisualization: FC<RecordingVisualizationProps> = ({
+  prevState,
+  state,
+  filePath,
+  basePath,
+  recItemName,
+}) => {
   const waveSurferRef = useRef<WaveSurfer>();
   const waveformContainerRef = useRef<HTMLDivElement>(null);
-  const waveformImageRef = useRef<HTMLImageElement>(null);
   const spectrogramContainerRef = useRef<HTMLDivElement>(null);
-  const spectrogramImageRef = useRef<HTMLImageElement>(null);
+  const [, , audioInputDeviceId, , ,] = useAudioInputOutputDevices();
+  const fileMonitor = new FileMonitor(basePath);
+
+  /**
+   * Check to see if the spectrogram plugin is active on the wavesurfer instance.
+   * If the plugin is not active, add and create a new instance of the Spectrogram plugin.
+   */
+  function checkSpectrogramPlugin() {
+    const waveSurfer = waveSurferRef.current;
+
+    if (!waveSurfer) {
+      return;
+    }
+
+    if (!('spectrogram' in waveSurfer.getActivePlugins())) {
+      waveSurfer
+        .addPlugin(
+          SpectrogramPlugin.create({
+            container: spectrogramContainerRef.current!,
+            labels: false,
+          }),
+        )
+        .initPlugin('spectrogram');
+    }
+  }
+
+  /**
+   * Load a .wav file into the waveform instance.
+   */
+  function loadWavFile(filePathToRead: string) {
```

Comment:
Use arrow function. Also declare return type

Reply:
when you say declare the return type, do you mean in the ts doc?

Reply:
```typescript
const functionName = (param1: Parameter1Type, param2: Parameter2Type): ReturnType => { /* Code omitted */ };
```